checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -419,16 +419,14 @@ protected void setSelfTypeInInitializationCode(
         // there might still be subclasses that need initialization.
         if (areAllFieldsCommittedOnly(enclosingClass)) {
             Store store = getStoreBefore(tree);
-            if (store != null) {
```

Comment:
Sorry, maybe this change should be undone, b/c it now adds the null check on line 497 below.
What I find strange about the null check below is that the nullness tests don't fail with an NPE, but with a completely unrelated assertion error.

Reply:
Oh, and I only run into this assertion violation when running the JUnit tests, not when manually compiling all the same files at once. So something is odd about this null check, too.

Reply:
The store was null only in a first pass through a constructor. I don't understand why the NPE would only show up in JUnit tests, not when run on an individual file. Something might still remain to be cleaned up here...